ui/lib/kv/addon/components/page/secret/details.js

```diff
@@ -90,24 +90,28 @@ export default class KvSecretDetails extends Component {
   async handleDestruction(type) {
     const { secret } = this.args;
     try {
-      await secret.destroyRecord({ adapterOptions: { deleteType: type, deleteVersions: secret.version } });
+      await secret.destroyRecord({ adapterOptions: { deleteType: type, deleteVersions: this.version } });
       this.store.clearDataset('kv/metadata'); // Clear out the store cache so that the metadata/list view is updated.
-      this.flashMessages.success(`Successfully ${secret.state} Version ${secret.version} of ${secret.path}.`);
+      this.flashMessages.success(`Successfully ${secret.state} Version ${this.version} of ${secret.path}.`);
       this.router.transitionTo('vault.cluster.secrets.backend.kv.secret', {
-        queryParams: { version: secret.version },
+        queryParams: { version: this.version },
       });
     } catch (err) {
       const verb = type.includes('delete') ? 'deleting' : 'destroying';
       this.flashMessages.danger(
-        `There was a problem ${verb} Version ${secret.version} of ${secret.path}. Error: ${err.errors.join(
+        `There was a problem ${verb} Version ${this.version} of ${secret.path}. Error: ${err.errors.join(
           ' '
         )}.`
       );
     }
   }
 
   get version() {
-    return this.args.secret.version || this.router.currentRoute.queryParams.version;
```

Comment:
dear goodness we can get version all over the place

Reply:
I know 😭